sensu/files/plugins/memory-metrics.rb

```diff
@@ -68,16 +68,20 @@ def metrics_hash
 
     mem['swapUsed'] = mem['swapTotal'] - mem['swapFree']
     mem['used'] = mem['total'] - mem['free']
-    memory_used = (100.0 * mem['used']) / mem['total']
+    mem['usedWOBuffersCaches'] = mem['used'] - (mem['buffers'] + mem['cached'])
+    mem['freeWOBuffersCaches'] = mem['free'] + (mem['buffers'] + mem['cached'])
+    mem['swapUsedPercentage'] = 100 * mem['swapUsed'] / mem['swapTotal'] if mem['swapTotal'] > 0
```

Comment:
WO stands for without, so
mem['freeWOBuffersCaches'] = mem['free'] + (mem['buffers'] + mem['cached']) means memory free without buffers and caches, or memory free if we don't include the memory that linux is using for disk caching. So, we add the memory free plus the memory being used by buffers and caches, because that memory is only being held for disk caching, but is available to the user if they need it.

and mem['usedWOBuffersCaches'] = mem['used'] - (mem['buffers'] + mem['cached']) is the memory used if we don't include the memory being used by the buffers and caches (because that memory isn't technically 'used', it's only being used by buffers and caches because the user doesn't need that memory for anything. Once the user needs that memory, linux will release it for the user).